torch/dispatch/dispatcher.py

```diff
@@ -0,0 +1,50 @@
+import torch
+import torch._C as _C
+from torch.utils._pytree import tree_flatten
+from torch.overrides import has_torch_function
+
+"""
+This is a dispatcher (in Python)
+- You can define new operations (in Python) without schemas
+- It interfaces with the PyTorch dispatcher
+"""
+
+class PyDispatcher:
+    def call(self, operator: "PyOperator", args, kwargs):
+        dispatch_key_set = compute_keyset(args, kwargs)
+        kernel = operator.lookup(dispatch_key_set)
+        return kernel(*args, **kwargs)
+
+    def redispatch(self, operator, dispatch_key_set, args, kwargs):
+        kernel = operator.lookup(dispatch_key_set)
+        return kernel(*args, **kwargs)
+
+
+dispatcher_singleton = PyDispatcher()
```

Comment:
It doesn't really seem like there's any reason to have a singleton object lol

Reply:
Yeah let me try to pull it